### PR TITLE
fix(TUP-26165)jar referenced in bat or sh classpath missing in build ZIP

### PR DESCRIPTION
**Pull request by jding-tlnd**
file folder lib
https://jira.talendforge.org/browse/TUP-26165

### CODE REVIEW

File: main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java [hunk 1]
```diff
@@ -725,6 +729,18 @@ protected void updateDependencySet(IFile assemblyFile) {
                 }
             }
         }
+        
+        try {
+            if (!routinesList.isEmpty()) {
```
Comment (zwxue) [inline main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java]: condition is not accurate here, some irrelevant user routine may have extra dependencies, should check dependencies in routine, beans, pigudf(<=7.2.1) for any case

File: main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java [hunk 2]
```diff
@@ -815,7 +831,17 @@ private void addToDuplicateLibs(Map<String, Set<Dependency>> map, Dependency dep
             Set<Dependency> set = new HashSet<>();
             map.put(coordinate, set);
         }
-        map.get(coordinate).add(dependency);
+        boolean exist = false;
+        for (Dependency dep : map.get(coordinate)) {
```
Comment (zwxue) [inline main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java]: just overwrite hasCode and equals of SortableDependeny is ok, no need to compare here

File: main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java (hunk 1 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java]: solved, thanks

File: main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java (hunk 2 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.maven/src/main/java/org/talend/designer/maven/tools/creator/CreateMavenJobPom.java]: solved, thanks

**Pull request merged**